### PR TITLE
Deprecate HttpClientDiscovery, use Psr18ClientDiscovery instead

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes
| Related tickets | -
| Documentation   | -
| License         | MIT

Relates to https://github.com/php-http/httplug/pull/171

### CODE REVIEW

File: src/Strategy/CommonClassesStrategy.php [hunk 1]
```diff
@@ -73,14 +73,14 @@ final class CommonClassesStrategy implements DiscoveryStrategy
...
         HttpAsyncClient::class => [
-            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, Promise::class, RequestFactory::class, [self::class, 'isPsr17FactoryInstalled']]],
+            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, Promise::class, [self::class, 'isPsr17FactoryInstalled']]],
             ['class' => Guzzle7::class, 'condition' => Guzzle7::class],
             ['class' => Guzzle6::class, 'condition' => Guzzle6::class],
             ['class' => Curl::class, 'condition' => Curl::class],
             ['class' => React::class, 'condition' => React::class],
         ],
         HttpClient::class => [
-            ['class' => SymfonyHttplug::class, 'condition' => [SymfonyHttplug::class, RequestFactory::class, [self::class, 'isPsr17FactoryInstalled']]],
```
Comment (nicolas-grekas) [inline src/Strategy/CommonClassesStrategy.php:83]: this is not needed, and will block when Symfony 7 will remove them from SymfonyHttplug

**Comment by hugofintecture** [comment]
Hello @nicolas-grekas! Can we use Psr18ClientDiscovery for `prependStrategy(MockClientStrategy::class);` ?
If yes, the example in the documentation is not up-to-date :) 
https://docs.php-http.org/en/latest/discovery.html?highlight=Psr18ClientDiscovery#mock-client-discovery

**Comment by nicolas-grekas** [comment]
Please send a doc PR :)

**Comment by hugofintecture** [comment]
Sure! Was to to be sure :)

**Pull request merged**